GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -545,53 +606,67 @@ static void StraightenLanes(int startIndex, int lastStraightenIndex, int lastLoo
             }
         }
 
-        private RevisionGraphRevision[] BuildOrderedNodesCache(int currentRowIndex)
+        /// <summary>
+        /// Build the ordered list of nodes from the revisions.
+        /// </summary>
+        /// <param name="currentRowIndex">The min node index for the row.</param>
+        /// <returns>The ordered revision cache.</returns>
+        private ImmutableArray<RevisionGraphRevision> BuildOrderedNodesCache(int currentRowIndex)
         {
-            if (_orderedNodesCache is not null && !_reorder && _orderedNodesCache.Length >= Math.Min(Count, currentRowIndex))
+            if (!_orderedNodesCacheInvalid && _orderedNodesCache.Length >= Math.Min(Count, currentRowIndex))
             {
                 return _orderedNodesCache;
             }
 
             // The Score of the referenced nodes can be rewritten by another thread.
-            // Then the cache is invalidated by setting _reorder.
+            // Then the cache is invalidated by setting _orderedNodesCache.
             // So if Sort() complains in this case, try again.
             while (true)
             {
                 try
                 {
-                    // Reset the reorder flag and the orderedUntilScore. This makes sure it isn't marked dirty before we even got to
-                    // rebuilding it.
-                    _orderedUntilScore = int.MinValue;
-                    _reorder = false;
```

Comment:
Found it right after writing the previous comment:
The necessary replacement for `_orderedUntilScore = int.MinValue;` is `_orderedNodesCache = ImmutableArray<RevisionGraphRevision>.Empty;`

Reply:
Good catch, I did not see that
Squashed existing, added this with updated commit message